### PR TITLE
Add `isDateOrDateTime`

### PR DESCRIPTION
**Pull request by ranquild**
It's a common need for the FE to check whether something is a date column. We don't usually care if it's just Date or DateTime. Currently we do that (incorrectly) by using either `isTemporal` or `isDateWithoutTime` functions. This PR introduces `isDateOrDateTime` function and fixes places where the other functions were used.

When I started to use `isDateOrDateTime` I faced issues that it returns `false` for date columns with extraction temporal units. This is expected. I tried to remove the temporal unit with `(lib/with-temporal-unit column nil)` but it didn't work and the effective type remained the same. This PR also fixes this issue by following what we do in `with-temporal-bucket` for `:field` - setting the `::original-effective-type` and using it when removing the temporal unit.

### CODE REVIEW

File: frontend/src/metabase/common/utils/columns.ts [hunk 1]
```diff
@@ -17,11 +17,7 @@ export function getColumnIcon(column: Lib.ColumnMetadata): IconName {
     return "location";
   }
 
-  if (
```
Comment (ranquild) [inline frontend/src/metabase/common/utils/columns.ts:20]: noop change

File: frontend/src/metabase-lib/viz/display.ts [hunk 2]
```diff
@@ -88,10 +88,10 @@ export const defaultDisplay = (query: Lib.Query): DefaultDisplay => {
       breakouts,
     );
 
-    const isAnyBreakoutDate = breakoutsWithColumns.some(({ column }) => {
+    const isAnyBreakoutTemporal = breakoutsWithColumns.some(({ column }) => {
```
Comment (ranquild) [inline frontend/src/metabase-lib/viz/display.ts:91]: just renaming

**Comment by camsaul** [comment]
Why don't you call it something like `isDateOrDateTime`. Sometimes it's important to differentiate dates versus datetimes, for example you don't want to show a time picker for a date column but it's ok for a datetime column. Naming this `isDate` seems like it could be potentially confusing there and lead to bugs in the future

**Comment by ranquild** [comment]
@camsaul agree, fixed

**Review by camsaul — APPROVED** [review]

File: src/metabase/lib/field.cljc [hunk 3]
```diff
@@ -378,7 +385,7 @@
 (defmethod lib.temporal-bucket/available-temporal-buckets-method :metadata/column
   [_query _stage-number field-metadata]
   (if (not= (:lib/source field-metadata) :source/expressions)
-    (let [effective-type ((some-fn :effective-type :base-type) field-metadata)
+    (let [effective-type ((some-fn ::original-effective-type :effective-type :base-type) field-metadata)
```
Comment (ranquild) [inline src/metabase/lib/field.cljc:388]: `with-temporal-bucket` + `available-temporal-buckets` is already tested for extraction units. When I fixed `effective-type` for such columns this place broke, so I had to fix it.

**Pull request merged**